core/dbt/context/runtime.py

```diff
@@ -47,13 +47,16 @@ def do_ref(*args):
                                            target_model_package)
 
         is_ephemeral = (get_materialization(target_model) == 'ephemeral')
-
         if is_ephemeral:
-            model.set_cte(target_model_id, None)
-            return adapter.Relation.create(
-                type=adapter.Relation.CTE,
-                identifier=add_ephemeral_model_prefix(
-                    target_model_name)).quote(identifier=False)
+            is_subquery = (get_ephemeral_type(target_model) == 'subquery')
+            if is_subquery:
+                return '(\n{}\n)'.format(target_model.compiled_sql)
+            else:
```

Comment:
can we do:
```
ephemeral_type = get_ephemeral_type(target_model)
if ephemeral_type == 'subquery':
..
elif ephemeral_type == 'cte' or ephemeral_type is None:
...
else:
  # raise an error here for an invalid value
```

You can raise an error with:
```
dbt.exceptions.raise_compiler_error(msg, target_model)
```

And the message can be something like:
> Invalid value for 'ephemeral_type'. Got '<value>' but expected on of 'cte' or 'subquery'